components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/WorkflowDAOImpl.java

```diff
@@ -127,12 +130,16 @@ public void updateWorkflowStatus(Workflow workflow) throws APIMgtDAOException {
     public Workflow retrieveWorkflow(String workflowReference) throws APIMgtDAOException {
         
         final String getworkflowQuery = "SELECT * FROM AM_WORKFLOWS WHERE WF_EXTERNAL_REFERENCE=?";
-        Workflow workflow;
+        Workflow workflow = null;
         try (Connection conn = DAOUtil.getConnection();
                 PreparedStatement ps = conn.prepareStatement(getworkflowQuery)) {
             ps.setString(1, workflowReference);
             try (ResultSet rs = ps.executeQuery()) {
-                workflow = this.createWorkflowFromResultSet(rs);
+                List<Workflow> list = this.createWorkflowFromResultSet(rs);
+                if (!list.isEmpty()) {
+                    workflow = list.get(0);
```

Comment:
For a given referenceId there is either one entry or no entry. That is why taking the first value. retrieveWorkflow(String workflowReference) method which calls createWorkflowFromResultSet() method passes resultset which only has a single entry.  I have re-used the generic createWorkflowFromResultSet() method which is used to generate list of workflow objects using the resultset which has more than one entry.